detection_rules/integrations.py

```diff
@@ -321,38 +319,78 @@ def get_integration_schema_data(data, meta, package_integrations: dict) -> Gener
-                                                                         rule_stack_version=min_stack,
-                                                                         packages_manifest=packages_manifest)
-
-                if notify_update_available and notice and data.get("notify", False):
-                    # Notify for now, as to not lock rule stacks to integrations
-                    notify_update_available = False
-                    print(f"\n{data.get('name')}")
-                    print(*notice)
-
-                schema = {}
-                if integration is None:
-                    # Use all fields from each dataset
-                    for dataset in integrations_schemas[package][package_version]:
-                        # ignore jobs from machine learning packages
-                        if dataset != "jobs":
-                            schema.update(integrations_schemas[package][package_version][dataset])
-                else:
-                    if integration not in integrations_schemas[package][package_version]:
-                        raise ValueError(f"Integration {integration} not found in package {package} "
-                                         f"version {package_version}")
-                    schema = integrations_schemas[package][package_version][integration]
-                schema.update(ecs_schema)
-                integration_schema = {k: kql.parser.elasticsearch_type_family(v) for k, v in schema.items()}
+                # Extract the integration schema fields
+                integration_schema, package_version = get_integration_schema_fields(integrations_schemas, package,
+                                                                                    integration, min_stack,
+                                                                                    packages_manifest, ecs_schema,
+                                                                                    data)
 
                 data = {"schema": integration_schema, "package": package, "integration": integration,
                         "stack_version": stack_version, "ecs_version": ecs_version,
                         "package_version": package_version, "endgame_version": endgame_version}
                 yield data
 
 
+def get_integration_schema_fields(integrations_schemas: dict, package: str, integration: str,
+                                  min_stack: Version, packages_manifest: dict,
+                                  ecs_schema: dict, data: dict) -> dict:
+    """Extracts the integration fields to schema based on package integrations."""
+
+    package_version, notice = find_latest_compatible_version(package, integration, min_stack, packages_manifest)
+    notify_user_if_update_available(data, notice, integration)
+
+    schema = collect_schema_fields(integrations_schemas, package, package_version, integration)
+    schema.update(ecs_schema)
```

Comment:
This means that for any colliding types, ECS schema takes precedence. I remember we discussed this during original implementation. There could also be implications further down this chain (see other comments)